modules/fuzzy/include/opencv2/fuzzy/fuzzy_F0_math.hpp

```diff
@@ -0,0 +1,119 @@
+// indirect, incidental, special, exemplary, or consequential damages
+// (including, but not limited to, procurement of substitute goods or services;
+// loss of use, data, or profits; or business interruption) however caused
+// and on any theory of liability, whether in contract, strict liability,
+// or tort (including negligence or otherwise) arising in any way out of
+// the use of this software, even if advised of the possibility of such damage.
+//
+//M*/
+
+#ifndef __OPENCV_FUZZY_F0_MATH_H__
+#define __OPENCV_FUZZY_F0_MATH_H__
+
+#include "types.hpp"
+#include "opencv2/core.hpp"
+
+namespace cv
+{
+
+namespace ft
+{
+    //! @addtogroup f0_math
+    //! @{
+
+    /** @brief Computes components of the array using direct F0-transform.
+    @param matrix Input 1-channel array.
+    @param kernel Kernel used for processing. Function **createKernel** can be used.
+    @param components Output 32-bit array for the components.
+    @param mask Mask can be used for unwanted area marking.
+
+    The function computes components using predefined kernel and mask.
+
+    @note
+        F-transform technique is described in paper @cite Perf:FT.
+     */
+    CV_EXPORTS void FT02D_components(cv::InputArray matrix, cv::InputArray kernel, cv::OutputArray components, cv::InputArray mask);
+
+    /** @brief Computes components of the array using direct F0-transform.
+    @param matrix Input 1-channel array.
+    @param kernel Kernel used for processing. Function **createKernel** can be used.
+    @param components Output 32-bit array for the components.
+
+    The function computes components using predefined kernel.
+
+    @note
+        F-transform technique is described in paper @cite Perf:FT.
+     */
+    CV_EXPORTS void FT02D_components(cv::InputArray matrix, cv::InputArray kernel, cv::OutputArray components);
+
+    /** @brief Computes inverse F0-transfrom.
+    @param components Input 32-bit array for the components.
+    @param kernel Kernel used for processing. Function **createKernel** can be used.
+    @param output Output 32-bit array.
+    @param width Width of the output array.
+    @param height Height of the output array.
+
+    @note
+        F-transform technique is described in paper @cite Perf:FT.
+     */
+    CV_EXPORTS void FT02D_inverseFT(cv::InputArray components, cv::InputArray kernel, cv::OutputArray output, int width, int height);
+
+    /** @brief Computes F0-transfrom and inverse F0-transfrom at once.
+    @param image Input image.
+    @param kernel Kernel used for processing. Function **createKernel** can be used.
+    @param output Output 32-bit array.
+    @param mask Mask used for unwanted area marking.
+
+    This function computes F-transfrom and inverse F-transfotm in one step. It is fully sufficient and optimized for **cv::Mat**.
+    */
+    CV_EXPORTS void FT02D_process(const cv::Mat &image, const cv::Mat &kernel, cv::Mat &output, const cv::Mat &mask);
```

Comment:
Well, I have a problem with this. I am focused on a performance in my implementation. I am trying to avoid any unnecessary *copyTo*.

With *InputArray*:
```cpp
void ft::FT02D_inverseFT(..., OutputArray output)
{
    ...

    output.create(..., ..., CV_32F);

    Mat outputZeroes(..., ..., CV_32F, Scalar(0));

    ...

    outputZeroes(Rect(..., ..., ..., ...)).copyTo(output);
}
```


This one (without *copyTo*) is not working (*outputMat* contains dummy values):
```cpp
void ft::FT02D_inverseFT(..., OutputArray output)
{
    ...

    Mat outputZeroes(..., ..., CV_32F, Scalar(0));

    ...

    Mat outputMat = output.getMat();
    outputMat = outputZeroes(Rect(..., ..., ..., ...));
}
```

but `outputMat = outputZeroes(Rect(..., ..., ..., ...));` is working correctly if I use *cv::Mat* instead of *InputArray*.

May i ask you, if my solution is ok in that case? Or is it possible to avoid expensive *copyTo* in case of *InputArray* usage?